lib/openscap_parser/value.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+module OpenscapParser
+  class Value < XmlNode
+    include OpenscapParser::Util
+
+    def id
+      @id ||= parsed_xml['id']
+    end
+
+    def description
+      @description ||= newline_to_whitespace(parsed_xml.at_css('description')&.text)
+    end
+
+    def title
+      @title ||= parsed_xml.at_css('title')&.text
+    end
+
+    def type
+      @type ||= parsed_xml['type'] || 'string'
+    end
+
+    def generic_selector(type, element, selector = nil)
+      unless instance_variable_get("@#{type}")
+        elements = parsed_xml.xpath(element).each_with_object({}) do |element, elements|
+          elements[element.at_xpath('@selector')&.text.presence] = element&.text
+        end
+        instance_variable_set("@#{type}", elements)
+      end
+
+      return instance_variable_get("@#{type}")[selector] if selector
+      instance_variable_get("@#{type}")[nil] || instance_variable_get("@#{type}").values.first
+    end
+
+    def upper_bound(selector = nil)
+      generic_selector(:upper_bound, 'upper-bound', selector)
```

Comment:
https://apidock.com/ruby/Symbol/to_s

Reply:
Do you mean to use this instead of passing the type and element name separately? Or instead of using the string interpolation here `instance_variable_get("@#{type}") `?

Reply:
I can't use `:upper_bound.to_s` to get the element name because the element name is `upper-bound` instead of `upper_bound`